bin/sandman/sweep.py

```diff
@@ -119,40 +127,127 @@ def _files(
+                    # attachement):
+                    if len(attachments[_key]) <= max_filelist_in_body:
+                        mail = StagedEMail(stakeholder, group_summary, n_files_key, file_list,
+                                           vault_documentation)
+                    else:
+                        mail = StagedEMail(stakeholder, group_summary, n_files_key, {},
+                                           vault_documentation)
+                        mail += GZippedFOFN(f"{_key}.fofn.gz",
+                                            [file.path for file in attachments[_key]])
+                    postman.send(mail, stakeholder)
+                    log.info(
+                        f"Sent E-mail for {_key} files to {stakeholder.name} ({stakeholder.email})")
+                else:
+                    log.debug(
+                        "Skipping E-mail for {_key} files to {stakeholder.name} ({stakeholder.email}): Trivial e-mail")
+
+                # E-mail notification (warning) for upcoming file deletions
+                for tminus in config.deletion.warnings:
+                    hours = int(time.seconds(tminus) / 3600)
+                    seconds = float(time.seconds(tminus))
+                    _key = f"delete-{hours}"
+                    group_summary = group_summaries[_key]
+                    file_list = file_lists[_key]
+                    n_files_key = n_files[_key]
+                    # TODO: write file list to lustre path revelant to
+                    # stakeholder
+                    filelist_lustre_path = ""
+                    if len(attachments[_key]) > 0:
+                        # Check if file list fits in e-mail body (instead of
+                        # attachement):
+                        if len(attachments[_key]) <= max_filelist_in_body:
+                            mail = WarnedEMail(stakeholder, group_summary, n_files_key,
+                                               file_list, vault_documentation,
+                                               seconds, filelist_lustre_path)
+                        else:
+                            mail = WarnedEMail(stakeholder, group_summary, n_files_key, {},
+                                               vault_documentation,
+                                               seconds, filelist_lustre_path)
+                            mail += GZippedFOFN(f"{filename}.fofn.gz",
+                                                [file.path for file in attachments[_key]])
+                            postman.send(mail, stakeholder)
+                            log.info(
+                                f"Sent E-mail warning for within {hours} hours upcoming file deletions to {stakeholder.name} ({stakeholder.email})")
+                    else:
+                        log.debug(
+                            "Skipping E-mail warning for within {hours} hours upcoming file deletions to {stakeholder.name} ({stakeholder.email}): Trivial e-mail")
```

Comment:
does feel like we're doing very similar things here, just with different constructors. could we do something like:
```python
emails = {'deletion': DeletionEmail, 'warning': WarningEmail, ...}
for _email_type, _EmailConstructor in emails.items():
    log.debug(f"I can use {_email_type} to do logging statements)
    mail = _EmailConstructor(...) # to create the right object
```

maybe not, the logic may be too different for the different types, I haven't fully read it

Reply:
yes I agree, the 3 constructors are very similar